calendar-bundle/contao/classes/Calendar.php

```diff
@@ -410,7 +395,14 @@ protected function addEvent($objEvent, $intStart, $intEnd, $strUrl, $strBase='',
 				break;
 
 			default:
-				$link = sprintf(preg_replace('/%(?!s)/', '%%', $strUrl), ($objEvent->alias ?: $objEvent->id));
+				$link = $objParent->getAbsoluteUrl('/' . ($objEvent->alias ?: $objEvent->id));
+
+				// Backwards compatibility (see #8329)
+				if ($strBase && !preg_match('#^https?://#', $link))
+				{
+					$link = $strBase . $link;
+				}
```

Comment:
I think we can remove this, because `getAbsoluteUrl()` will always return an absolute URL now.